File: app/src/main/java/com/pbt/cogni/activity/chat/adapter/DownloadFileFromURLTask.java
package com.pbt.cogni.activity.chat.adapter;

import android.annotation.SuppressLint;
import android.os.AsyncTask;
import android.os.Environment;
import android.util.Log;

import java.io.BufferedInputStream;
import java.io.File;
import java.io.FileOutputStream;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.URL;
import java.net.URLConnection;

class DownloadFileFromURfLTask extends
        AsyncTask<String, String, String> {

    /**
     * Downloading file in background thread
     * */
    @SuppressLint("SdCardPath")
    @Override
    protected String doInBackground(String... f_url) {
        Log.d("##ChatAdapter", "do in background");
        int count;
        try {
            URL url = new URL(f_url[0]);
            URLConnection conection = url.openConnection();
            conection.connect();
            // getting file length
            int lenghtOfFile = conection.getContentLength();

            // input stream to read file - with 8k buffer
            InputStream input = new BufferedInputStream(url.openStream(),
                    8192);
            File myObj = new File(Environment.getExternalStorageDirectory()+"/"+"filename.txt");
            // Output stream to write file
            OutputStream output = new FileOutputStream(myObj.toString());

            byte data[] = new byte[1024];

            long total = 0;

            while ((count = input.read(data)) != -1) {
                total += count;
                // publishing the progress....
                // After this onProgressUpdate will be called
                publishProgress("" + (int) ((total * 100) / lenghtOfFile));

                // writing data to file
                output.write(data, 0, count);
            }

            // flushing output
            output.flush();

            // closing streams
            output.close();
            input.close();

        } catch (Exception e) {
            Log.e("Error: ", e.getMessage());
        }

        return null;
    }

    /**
     * Updating progress bar
     * */
    protected void onProgressUpdate(String... progress) {
        // setting progress percentage

        Log.d("##DownloadFile",progress.toString());




    }

    /**
     * After completing background task Dismiss the progress dialog
     * **/

    @Override
    protected void onPostExecute(String file_url) {
        Log.i("##ChatAdapter", "on post excute!");


    }


}

File: app/src/main/java/com/pbt/cogni/activity/chat/adapter/DownloadFileFromURLTask.kt
package com.pbt.cogni.activity.chat.adapter

import android.annotation.SuppressLint
import android.os.AsyncTask
import android.os.Environment
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import androidx.core.content.ContextCompat
import androidx.databinding.DataBindingUtil
import com.pbt.cogni.R
import com.pbt.cogni.activity.chat.ChatActivity
import java.io.*
import java.net.URL
import java.security.AccessController.getContext


class DownloadFileFromURLTask() : AsyncTask<String?, String?, String?>() {




    /**
     * Downloading file in background thread
     */
    @SuppressLint("SdCardPath")
    override fun doInBackground(vararg f_url: String?): String? {
        Log.d("##ChatAdapter", "do in background")
        var count: Int
        try {
            val url = URL(f_url[0])
            val conection = url.openConnection()
            conection.connect()
            // getting file length
            val lenghtOfFile = conection.contentLength

            // input stream to read file - with 8k buffer
            val input: InputStream = BufferedInputStream(
                url.openStream(),
                8192
            )
            val myObj =
                File(Environment.getExternalStorageDirectory().toString() + "/" + "filename.txt")
            // Output stream to write file
            val output: OutputStream = FileOutputStream(myObj.toString())
            val data = ByteArray(1024)
            var total: Long = 0
            while (input.read(data).also { count = it } != -1) {
                total += count.toLong()
                // publishing the progress....
                // After this onProgressUpdate will be called
                publishProgress("" + (total * 100 / lenghtOfFile).toInt())

                // writing data to file
                output.write(data, 0, count)
            }

            // flushing output
            output.flush()

            // closing streams
            output.close()
            input.close()
        } catch (e: Exception) {
            Log.e("Error: ", e.message)
        }
        return null
    }

    /**
     * Updating progress bar
     */
    override fun onProgressUpdate(vararg progress: String?) {
        // setting progress percentage
        Log.d("##DownloadFile", progress.toString())
//        progresss?.setProgress(Integer.parseInt(progress[0]!!))

      ChatActivity.progressbar?.setProgress(Integer.parseInt(progress[0]!!))
        Log.d("##DownloadFile", ChatActivity.progressbar?.progress.toString())

    

    }

    /**
     * After completing background task Dismiss the progress dialog
     */
    override fun onPostExecute(file_url: String?) {
        Log.i("##ChatAdapter", "on post excute!")
    }


}